src/isotools/_transcriptome_stats.py

```diff
@@ -723,24 +723,27 @@ def _filter_event(event, transcriptome, gene, min_total=100, min_alt_fraction=.1
     return True
 
 
-def pairwise_event_test(e1, e2, transcriptome, gene, min_dist=1, test="chi2"):
+def pairwise_event_test(e1, e2, transcriptome, gene, min_dist=1, test="chi2", **kwargs):
     '''
     performs an independence test among the joint frequencies of two events
 
     :param e1: event obtained from .find_splice_bubbles()
     :param e2: event obtained from .find_splice_bubbles()
-    :param transcriptome: the Transcriptome object on which the events were computed 
+    :param transcriptome: the Transcriptome object on which the events were computed
     ;param gene: Ensemble ID of the gene in which the event happened
     :type gene: str
     :param min_dist: minimum distance (in nucleotides) between the two Alternative Splicing Events for the pair to be tested
     :type min_dist: int
     :param test: test to be performed. One of ("chi2", "fisher")
     :type test: str
+    :param sg: segment graph
     '''
 
+    sg = kwargs["sg"] if "sg" in kwargs else transcriptome[gene].segment_graph
+
     coverage = transcriptome[gene].coverage.sum(axis=0)
 
-    if events_dist(e1, e2) > min_dist:
+    if events_dist(e1, e2, sg) > min_dist:
```

Comment:
Do that check before calling the function

Reply:
Ok. Moved the check before the function call.